migrations/20241105164503_2.sql

```diff
@@ -0,0 +1,40 @@
+-- Force anything running this migration to use the right search path.
+set local search_path to underway;
+
+-- Remove attempt-specific columns from `task`
+alter table underway.task
+drop column if exists retry_count,
+drop column if exists initial_interval_ms,
+drop column if exists max_interval_ms,
+drop column if exists backoff_coefficient,
+drop column if exists error_message,
+drop column if exists started_at,
+drop column if exists succeeded_at,
+drop column if exists last_failed_at;
+
+alter table underway.task
+add column if not exists retry_policy jsonb not null default '{"max_attempts": 5, "initial_interval_ms": 1000, "max_interval_ms": 60000, "backoff_coefficient": 2.0}';
+
+alter table underway.task
+add column if not exists completed_at timestamp with time zone;
+
+alter table underway.task
+add column if not exists last_attempt_at timestamp with time zone;
+
+create table underway.task_attempt (
+    task_id         uuid not null,
+    task_queue_name text not null,
+    attempt_number  integer not null,
+
+    -- Task state.
+    state           underway.task_state not null default 'pending',
+
+    -- Error metatdata.
+    error_message   text,
+
+    created_at      timestamp with time zone not null default now(),
```

Comment:
Correct me if I'm wrong but I think those are always derivable:

- `started_at` is always `created_at` (since an attempt is created upon dequeue)
- `completed_at` is `updated_at` when state is "succeeded" or "failed"

We could rename `created_at` if that's more clear and technically we could set a separate `completed_at` column to be sure `updated_at` isn't overloaded unnecessarily.

Reply:
you are correct, my bad

Reply:
Do you think it's worth having a separate `completed_at` column? I can't immediately think of use cases for changing `updated_at` after an attempt is completed, but it would ensure that the semantic isn't conflated.

Reply:
I would prefer `started_at/created_at`

Reply:
Great I'll update that too: I think this is `started_at` and `completed_at`, correct?

Reply:
I've added this now.